source/System/Text/StringBuilder.cs

```diff
@@ -364,7 +364,7 @@ public StringBuilder Append(long value)
         /// <returns>A reference to this instance after the append operation has completed.</returns>
         public StringBuilder Append(object value)
         {
-            return value == null ? this : Append(value);
+            return Append(value.ToString());
```

Comment:
The check for null is still required, otherwise the ToString() would fail.

Check [here](https://referencesource.microsoft.com/#mscorlib/system/text/stringbuilder.cs,1002) the code from the full framework.

 
Please follow.